### ISSUE TITLE
[CM P0] Revitalize webd and elektrad

### ISSUE THREAD

**Issue by lukashartl** [ElektraInitiative/libelektra#4299]
# Introduction

For our project we would like to revitalize elektra's web-components:

* `web-ui` (Instance Manager, Frontend, React)
* `webd` (Instance Manager, Backend, Express)
* `elektrad` (Instance, Backend, written in Go)

Introduction to those components can be found here: https://www.libelektra.org/tutorials/webui

As `elektrad` is also currently missing an active maintainer, this is also a great chance to update it's dependencies and such.

This issue serves as our `main-ticket`, where we will link all other pull requests and keep track of the tasks.

# Quick Start

```sh
mkdir build
cd build/
cmake .. -DTOOLS="kdb;web" -DPLUGINS="ALL;-EXPERIMENTAL" -DBINDINGS="ALL;-EXPERIMENTAL"
make -j2
sudo make install

(1st shell) kdb run-elektrad
(2nd shell) INSTANCE="http://localhost:33333/" kdb run-webd
```

* `elektrad` will listen on <http://localhost:33333/>
* `webd` will listen on <http://localhost:33334/>

# Related Issues

- #2441
- #2579: workaround for #2441
- #4263
- #2913
- #4277
- #4317

# Current state of affairs

- The keyname overhaul introduced the namespace syntax: `namespace:/path/to/key` and the javascript parts haven't been updated to incorporate these changes.
- `elektrad` and `webd` start without errors

Problems:

- [x] Each namespace is currently shown twice. one time with the new namespace syntax and once with only the namespace name. (fixed in #4353)
  - [x] the new syntax is expandable, but will raise errors
  - [x] the old syntax is not expandable
- [x] when expanding namespaces:
  - [x] `invalid json response body at http://localhost:33333/kdb/spec%3A?preload=1 reason: Unexpected end of JSON input`
  - [x] due to incorrect namespace use
- [x] when expanding Keys:
  - [x] `Error: invalid response from server`
  - [x] Missing Set-Cookie Header is not handled correctly
- [x] Search is not working
  - [x] Missing Set-Cookie Header is not handled correctly

**Comment by lukashartl** [comment]
I also found the following PRs made by dependabot. Those should be easy to review. Mostly they are just upgrading the patch-number (MAJOR.MINOR.PATCH).

**webd (Instance Manager, Backend, Express)**:

* #4339
* #4296

**web-ui (Instance Manager, Frontend, React)**:

* #4332
* #4326

/cc @leothetryhard

**Comment by lukashartl** [comment]
I closed both #4332 and #4296 , because we did already update those dependencies in #4353 .

**Comment by lukashartl** [comment]
further issues are tracked in #4356
